cpp/include/cudf/table/row_operators.cuh

```diff
@@ -343,6 +358,8 @@ class row_lexicographic_comparator {
   {
     // Add check for types to be the same.
     CUDF_EXPECTS(_lhs.num_columns() == _rhs.num_columns(), "Mismatched number of columns.");
+    CUDF_EXPECTS(detail::is_relationally_comparable(_lhs, _rhs),
```

Comment:
You can remove the comment above about checking the types to be the same since that's effectively what `is_relationally_comparable` does.